src/grpc/grpc_server.cc

```diff
@@ -1414,7 +1417,31 @@ CommonHandler::RegisterTrace()
     (*response->mutable_settings())["log_frequency"].add_value(
         std::to_string(log_frequency));
     (*response->mutable_settings())["trace_file"].add_value(filepath);
-
+    (*response->mutable_settings())["trace_mode"].add_value(
+        trace_manager_->InferenceTraceModeString(trace_mode));
+    {
+      auto mode_key = std::to_string(trace_mode);
+      auto trace_options_it = config_map.find(mode_key);
+      if (trace_options_it != config_map.end()) {
+        for (const auto& element : trace_options_it->second) {
+          if ((element.first == "file") || (element.first == "log-frequency")) {
+            continue;
+          }
+          std::string valueAsString;
+          if (std::holds_alternative<std::string>(element.second)) {
+            valueAsString = std::get<std::string>(element.second);
+          } else if (std::holds_alternative<int>(element.second)) {
+            valueAsString = std::to_string(std::get<int>(element.second));
+          } else if (std::holds_alternative<uint32_t>(element.second)) {
+            valueAsString = std::to_string(std::get<uint32_t>(element.second));
+          }
```

Comment:
Can these `if`/`else` statements be condensed to a single `std::to_string(element.second)` for all cases? or do we need the intermediate `std::get` conversions for each case?
```cpp
          const auto valueAsString = std::to_string(element.second);
          (*response->mutable_settings())[element.first].add_value(
              valueAsString);
```

Reply:
Thanks for looking into this, agreed on readabiity, I'm good with how it currently is.